src/Model/Timestampable/TimestampableCurrentDateTimeMethodTrait.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Knp\DoctrineBehaviors\Model\Timestampable;
+
+use DateTime;
+use DateTimeInterface;
+use DateTimeZone;
+use Knp\DoctrineBehaviors\Exception\ShouldNotHappenException;
+
+trait TimestampableCurrentDateTimeMethodTrait
+{
+    /**
+     * Returns a DateTimeInterface object with the current timestamp.
+     *
+     * @throws ShouldNotHappenException
+     */
+    public function getCurrentDateTime(): DateTimeInterface
+    {
+        // Create a datetime with microseconds.
+        $dateTime = DateTime::createFromFormat('U.u', sprintf('%.6F', microtime(true)));
```

Comment:
@TomasVotruba Btw. I have been wondering about this code. Why is it so complicated.
A PHP DateTime object contains microseconds if I create it like  `new DateTime()`, is already assigned the default timezone and the current datetime (with microseconds). At least since PHP 7.1 and required PHP is 7.2 or higher.